src/state/swap/index.ts

```diff
@@ -59,23 +52,20 @@ export type OnTokenChange = (field: Field, token: Currency) => void
 /** An integration hook called when the user enters a new amount. */
 export type OnAmountChange = (field: Field, amount: string) => void
 
-/**
- * An integration hook called when the user switches the tokens.
- * The values represent already-switched state, to make it easier to update controlled state.
- */
-export type OnSwitchTokens = (values: SwapController) => void
+/** An integration hook called when the user switches the tokens. */
+export type OnSwitchTokens = () => void
 
 /**
  * An integration hook called when the user clicks 'Review swap'.
  * If the hook resolves to false or rejects, the review dialog will not open.
  */
-export type OnReviewSwapClick = () => void | Promise<boolean>
+export type OnReviewSwapClick = () => void | boolean | Promise<boolean>
 
 /**
  * An integration hook called when the user clicks the token selector.
  * If the hook resolve to false or rejects, the token selector will not open.
  */
-export type OnTokenSelectorClick = (field: Field) => void | Promise<boolean>
+export type OnTokenSelectorClick = (field: Field) => void | boolean | Promise<boolean>
```

Comment:
can you explain why this change is desirable?

Reply:
In the case that you always want to disable token selection, it's easier to write `() => false` than `() => Promise.resolve(false)`.